code/obj/storage/secure_closets.dm

```diff
@@ -4,6 +4,95 @@
 	soundproofing = 5
 	can_flip_bust = 1
 	p_class = 3
+	_max_health = LOCKER_HEALTH_AVERAGE
+	_health = LOCKER_HEALTH_AVERAGE
+	/// Anchored if TRUE
+	var/bolted = TRUE
+	/// Can't be broken open with melee
+	var/reinforced = FALSE
+
+	New()
+		..()
+		if (bolted)
+			anchored = 1
+
+	get_desc(dist)
+		. += "[reinforced ? "It's reinforced, only a bomb could break into this. " : ""] [bolted ? "It's bolted to the floor." : ""]"
+
+	attackby(obj/item/I as obj, mob/user as mob)
+		if (src.open || !src.locked)
+			..()
+		else if (!I)
+			..()
+		else if (istype(I, /obj/item/satchel/))
+			..()
+		else if (isweldingtool(I))
+			..()
+		else if (iswrenchingtool(I))
+			if (istype(get_turf(src), /turf/space))
+				if (user)
+					user.show_text("What exactly are you gunna secure [src] to?", "red")
+				return
+			playsound(src.loc, "sound/items/Ratchet.ogg", 50, 1)
+			SETUP_GENERIC_ACTIONBAR(user, src, 5 SECONDS, .proc/toggle_bolts, user, I.icon, I.icon_state,"", null)
+		else if (istype(I, /obj/item/card/))
+			..()
+		else if (user.a_intent == INTENT_HELP)
+			..()
+		else if ((I.force > 0) && !reinforced)
+			user.visible_message("<span class='alert'><b>[user]</b> hits [src] with [I]!</span>")
+			user.lastattacked = src
+			attack_particle(user,src)
+			hit_twitch(src)
+			take_damage(clamp(I.force, 1, 20), user)
+			playsound(src.loc, 'sound/impact_sounds/locker_hit.ogg', 90, 1)
+		else
+			..()
+
+	bullet_act(var/obj/projectile/P)
+		var/damage = 0
+		if (!P || !istype(P.proj_data,/datum/projectile/))
+			return
+		if (reinforced)
+			// To prevent parking in front of NARCS for ammo free locker entry
+			if (istype(P.proj_data, /datum/projectile/bullet/abg))
+				return
+		damage = round((P.power*P.proj_data.ks_ratio), 1.0)
+		// .22 is quite high raw damage and can be manufactured. Custom is material dipped .22. Stinger grenades did 60 damage.
+		if (istype(P.proj_data, /datum/projectile/bullet/bullet_22) || istype(P.proj_data, /datum/projectile/bullet/custom) || istype(P.proj_data, /datum/projectile/bullet/stinger_ball))
+			damage = round(damage * 0.5)
```

Comment:
typechecking on projectiles feels really weird and I don't like it

Reply:
Would a var be preferable?
Calibre has the issue of being shared.

Reply:
could do a check on projectile power - all 3 of those projectiles are <25 power, for example